ginivision/src/main/java/net/gini/android/vision/GiniVisionBasePresenter.java

```diff
@@ -0,0 +1,34 @@
+package net.gini.android.vision;
+
+import android.app.Application;
+import android.support.annotation.NonNull;
+
+/**
+ * Created by Alpar Szotyori on 08.05.2019.
+ *
+ * Copyright (c) 2019 Gini GmbH.
+ */
+public abstract class GiniVisionBasePresenter<V extends GiniVisionBaseView> {
+
+    private final Application mApp;
+    private final V mView;
```

Comment:
I don't think so. The View owns the Presenter instance (View creates the Presenter). They have a mutual reference, but the garbage collector can handle this. When there is no more reference to the View it will collect it and also collect any object which is referenced by the View, including the Presenter.

The reference to the Application is fine since it is an outward reference and won't prevent gc of the Presenter.

Reply:
To be on the safe side I will ✅ ✅ this.

Reply:
Checked it with the profiler. There is no memory leak and presenter, view and interactor are all garbage collected 🎉 .

Reply:
Awesome, good think that things are handle by the GC 💯